# Correlation Analysis Project MCAS


# Introduction
We are looking at data from Massachuchets public high schools from 1998.
We would like to know what are the 3 things that correlate highest with 4th grade test scores. We found that the three top corelating where % Eligible for free/reduced price lunch, Per Capita Income, and Percent English Learners. Some with low correlations that where noteworthy was the data on spending per student. 



# Dataset Preparation 
The dataset has alot of data on Massachuchets public high schools from 1998. It was given to us from Tracey but the data was origanally taken from the Massachusetts Comprehensive Assessment System (MCAS) test. Below is the code used to prepare the data analysis. It imports the needed python packages and puts the data from the csv file into a dataframe.


In [6]:
import csv
import math
import statistics as stats
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt


mcas = pd.read_csv('MCAS.csv') #read in file to dataframe

mcas[:3]

Unnamed: 0  code  municipa  district  regday     specneed  bilingua  \
0           1     1  ABINGTON  Abington    4201  7375.689941         0   
1           2     2     ACTON     Acton    4129  8573.990234         0   
2           3     3  ACUSHNET  Acushnet    3627  8081.720215         0   

   occupday  totday   spc  speced  lnchpct   tchratio  percap  totsc4  totsc8  \
0         0    4646  16.6    14.6     11.8  19.000000  16.379     714   691.0   
1         0    4930   5.7    17.4      2.5  22.600000  25.792     731     NaN   
2         0    4281   7.5    12.1     14.1  19.299999  14.040     704   693.0   

   avgsalary     pctel  
0  34.360001  0.000000  
1  38.063000  1.246106  
2  32.491001  0.000000

# Data Modeling
- We wanted to compare every quantitative variable to 4th grade test scores in order extrapalate early test scores 
- I used pearson coefficiants from the numpy packages in python to calculate how corelated each section was with 4th grade test score


In [7]:
#create a list of correlation coeffcients for minutes 0 - 47
r_vals = [] 

#loop through columns and compute Pearson correlation coefficient
for c in mcas.columns:
    #skip the team name column and the minute 48 column (its our anchor variable)
    if c != 'Column1' and c != 'code' and c != 'municipa' and c != 'district' and c != 'totsc4' and c != 'spc' and c != 'totsc8' and c != 'avgsalary':
        #calculate the coefficient and append it to our list
        r = np.corrcoef(mcas[c], mcas['totsc4'])[0, 1]
        r_vals.append(r) 
        print("Correlation in ", c, " is ", r)
        
        


Correlation in  Unnamed: 0  is  0.0187515854994
Correlation in  regday  is  0.179493063031
Correlation in  specneed  is  0.0931881628459
Correlation in  bilingua  is  -0.100864012695
Correlation in  occupday  is  -0.418619464134
Correlation in  totday  is  0.108952042221
Correlation in  speced  is  -0.241002372748
Correlation in  lnchpct  is  -0.78418370215
Correlation in  tchratio  is  -0.258545334661
Correlation in  percap  is  0.623420563566
Correlation in  pctel  is  -0.52797467598


# Data Analysis
- After looking at all the r values from the data sets, I've found that the three most corelated to totsc4 are lnchpct r = -0.78418370215 percap r = 0.623420563566 pctel r = -0.52797467598
    - The three highest corelating, in order where
        - % Eligible for free/reduced price lunch
        - Per Capita Income
        - Percent English Learners
    - Note they had a negative relation other than the per capita income


- Some of the lowest corelating factors with 4th grade test scores are spending per pupil, specifically for regular, special needs, bilingual, and total
    - Note that we dont know how these tests where administered so it is possible that the data especially for special needs and bilingial students is skewed
    - If we where to ignore possible failures in the data gathering we could say that we could reduce spending in schools and achieive the same test scores for our students, but it is important to note that test scores may not be the perfect way to messure the value of someones education
        - If we did more analysis this would be a topic to dive deeper into 




# Acknowledgements
Used alot of Traceys code